blueflood-core/src/test/java/com/rackspacecloud/blueflood/inputs/processors/IncomingMetricMetadataAnalyzerTest.java

```diff
@@ -0,0 +1,79 @@
+package com.rackspacecloud.blueflood.inputs.processors;
+
+import com.rackspacecloud.blueflood.cache.MetadataCache;
+
+import com.rackspacecloud.blueflood.io.MetadataIO;
+import com.rackspacecloud.blueflood.service.IncomingMetricMetadataAnalyzer;
+import com.rackspacecloud.blueflood.types.*;
+import com.rackspacecloud.blueflood.utils.InMemoryMetadataIO;
+import com.rackspacecloud.blueflood.utils.TimeValue;
+import junit.framework.Assert;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.concurrent.TimeUnit;
+
+public class IncomingMetricMetadataAnalyzerTest {
+    private final MetadataIO metadataIO;
+
+    public IncomingMetricMetadataAnalyzerTest () {
+        this.metadataIO = new InMemoryMetadataIO();
+    }
+
+    @Test
+    public void test_ScanMetrics_DoesNotStoreTypeAndUnit_ForNumericMetrics() throws Exception {
+        MetadataCache cache = MetadataCache.createLoadingCacheInstance(new TimeValue(5, TimeUnit.MINUTES), 1);
+        cache.setIO(this.metadataIO);
+        IncomingMetricMetadataAnalyzer analyzer = new IncomingMetricMetadataAnalyzer(cache);
+
+        ArrayList<IMetric> metricsList = new ArrayList<IMetric>();
+
+        Locator locator = Locator.createLocatorFromDbKey("integer_metric");
+        metricsList.add(new Metric(locator, 123, 14200000, new TimeValue(1, TimeUnit.DAYS),"somethings"));
+
+        analyzer.scanMetrics(metricsList);
+        String type = cache.get(locator, MetricMetadata.TYPE.name().toLowerCase());
+        Assert.assertEquals(null, type);
+
+        String unit = cache.get(locator, MetricMetadata.UNIT.name().toLowerCase());
```

Comment:
This test will be changed, once you remove the `checkMeta` for unit in `IncomingMetadataAnalyzer` code.